### ISSUE TITLE
probabilistic matrix factorization

### ISSUE THREAD

**Issue by OriolAbril** [pymc-devs/pymc-examples#58]
File: https://github.com/pymc-devs/pymc-examples/blob/main/examples/case_studies/probabilistic_matrix_factorization.ipynb
Reviewers: @colcarroll

## Known changes needed
Changes listed in this section should all be done at some point in order to get this
notebook to a "Best Practices" state. However, these are probably not enough!
Make sure to thoroughly review the notebook and search for other updates.

### General updates
* Use numpy [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator). See also https://numpy.org/doc/stable/reference/random/index.html?highlight=random%20sampling%20numpy%20random#quick-start

## Changes for discussion
Changes listed in this section are up for discussion, these are ideas on how to improve
the notebook but may not have a clear implementation, or fix some know issue only partially.

### ArviZ related
* Use ArviZ and xarray for postprocessing. This will probably be challenging. I'd recommend familiarizing with xarray before working on that. Some ideas:
   * `_norms` in code cell 23 looks like it could be replaced by `xr.apply_ufunc` (using `input_core_dims`)
### Exotic dependencies
None

### Computing requirements
Model samples in roughly 1 hour

**Comment by OriolAbril** [comment]
Note: the "good first issue" label only applies to updating the notebook from `To Do` to `General Updates`.

**Comment by chiral-carbon** [comment]
Hi @OriolAbril , can I take this issue up? I am going through the probabilistic matrix factorization notebook at the moment and thought I could attempt to make the recommended changes.

**Comment by OriolAbril** [comment]
Definitely,  I'll assign it to you. Let me know if you have any doubts, especially on xarray which can be challenging even for seasoned pymc3 users

**Comment by chiral-carbon** [comment]
sure, thanks! I'll get back to you on this.
